gravitee-apim-plugin/gravitee-apim-plugin-endpoint/gravitee-apim-plugin-endpoint-mqtt5/src/main/java/io/gravitee/plugin/endpoint/mqtt5/Mqtt5EndpointConnector.java

```diff
@@ -214,16 +195,59 @@ protected Completable publish(ExecutionContext ctx) {
         }
     }
 
-    private String getIdentifier(ExecutionContext ctx) {
-        String identifier = ctx.getAttribute(CONTEXT_ATTRIBUTE_MQTT5_IDENTIFIER);
-        if (identifier == null || identifier.isEmpty()) {
-            identifier = configuration.getIdentifier();
-            if (identifier == null) {
-                identifier = ctx.request().transactionId();
-                ctx.setAttribute(CONTEXT_ATTRIBUTE_MQTT5_IDENTIFIER, identifier);
-            }
+    private String generatePublisherIdentifier() {
+        return "gio-apim-publisher-" + UUID.randomUUID().toString().split("-")[0];
+    }
+
+    protected Mqtt5SubscribeBuilder.Publishes.Start.Complete<FlowableWithSingle<Mqtt5Publish, Mqtt5SubAck>> prepareMqtt5Subscribe(
+        final ExecutionContext ctx,
+        final Mqtt5RxClient rxClient,
+        final String topic
+    ) {
+        return rxClient.subscribePublishesWith().topicFilter(topic).qos(Mqtt5Publish.DEFAULT_QOS);
+    }
+
+    protected MqttConnectBuilder.Default prepareMqttConnect() {
+        return MqttConnect.DEFAULT.extend();
+    }
+
+    protected Mqtt5ClientBuilder prepareMqtt5Client(final ExecutionContext ctx) {
+        return Mqtt5Client
+            .builder()
+            .serverHost(configuration.getServerHost())
+            .serverPort(configuration.getServerPort())
+            .automaticReconnectWithDefaultConfig()
+            .addDisconnectedListener(
+                context -> {
+                    if (context.getReconnector().getAttempts() >= configuration().getReconnectAttempt()) {
+                        context.getReconnector().reconnect(false);
+                    }
+                }
+            );
+    }
+
+    protected Mqtt5PublishBuilder.Complete prepareMqtt5Publish(
+        final ExecutionContext ctx,
```

Comment:
unused :)

Reply:
unused here but useful somewhere else for example in the mqtt advanced.

Reply:
yes, I did not review the advanced plugin yet when writing this comment and forgot to remove it